iroh/src/services.rs

```diff
@@ -252,8 +252,12 @@ where
     Ok(())
 }
 
-/// require a set of services is up
-pub async fn require_services(api: &impl Api, services: HashSet<&str>) -> Result<()> {
+/// require a set of services is up. returns the underlying status table of all
+/// services for additional scrutiny
+pub async fn require_services(
+    api: &impl Api,
+    services: HashSet<&str>,
```

Comment:
I would prefer BTreeSet in such cases because it is more deterministic. But in this case it does not really matter.

Reply:
yeah totally. I'll come through & refactor all uses of `HashSet` to `BTreeSet` in a follow up